scrape/target.go

```diff
@@ -72,6 +74,23 @@ func (t *Target) String() string {
 	return t.URL().String()
 }
 
+// Metadata returns type and help metadata for the given metric.
+func (t *Target) Metadata(metric string) (textparse.MetricType, string, bool) {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	if t.metadataFunc == nil {
+		return textparse.MetricTypeUnknown, "", false
+	}
+	return t.metadataFunc(metric)
```

Comment:
You probably want to either unlock before calling this function or wrap with RLock(), or concurrency for this API will effectively be limited to 1.

If you unlock, consider switching the RWMutex to a regular Mutex or an atomic pointer, which are lighter options for this use.

Reply:
Never mind, the concurrency limit is for the target, not the API, so probably not important. I think it's still cleaner to either call RLock or switch to Mutex.

Reply:
Yes, that makes sense.